lib/reservation_manager.rb

```diff
@@ -0,0 +1,136 @@
+require_relative "reservation.rb"
+require "date"
+
+require "awesome_print"
+
+module Hotel
+  class ReservationManager
+    #add constant for room numbers
+    MAX_ROOMS = 20
+    ROOM_RATE = 200
+
+    def initialize
+      @reservations = []
+      @block = []
+    end
+
+    def all_rooms
+      rooms = (1..MAX_ROOMS).to_a
+      return rooms
+    end
+
+    def make_reservation(check_in, check_out)
+      # unsure how to raise argument error when room is unavailable since room_number is assigned from avail_rooms array
+      if available_rooms(check_in, check_out).length == 0
+        raise ArgumentError, "There are no available rooms for this time period"
+      else
+        room_number = available_rooms(check_in, check_out).first
+        # reservation_rate = reservation_cost(check_in, check_out)
+
+        new_reservation = Hotel::Reservation.new(room_number,
+                                                 check_in,
+                                                 check_out)
+      end
+      @reservations << new_reservation
+
+      return @reservations
+    end
+
+    def reservations_by_date(start_date, end_date)
+      s_date = Date.parse(start_date)
+      e_date = Date.parse(end_date)
+      reservations_by_date = []
+      @reservations.each do |reservation|
+        if e_date > reservation.check_in && e_date <= reservation.check_out
+          reservations_by_date << reservation
+        elsif s_date >= reservation.check_in && s_date < reservation.check_out
+          reservations_by_date << reservation
+        elsif s_date <= reservation.check_in && e_date >= reservation.check_out
```

Comment:
Nice work on finding a way to make all of this logic readable, even though it's weird and complex :) I think there's an opportunity to pull some of this out into different methods or classes, but I wouldn't push for it if it doesn't seem high priority